releaser/src/releaser.rs

```diff
@@ -0,0 +1,227 @@
+
+pub struct Releaser {
+    packages: PackageList,
+    git: Git,
+}
+
+pub fn ensure_repo_ready(git: &Git) -> Result<()> {
+    if !git.clean()? {
+        anyhow::bail!("Repository is not clean");
+    }
+
+    let branch = git.get_branch()?;
+    if branch != "main" {
+        anyhow::bail!("Git branch is not main");
+    }
+
+    Ok(())
+}
+
+impl Releaser {
+    pub fn new(packages: PackageList) -> Self {
+        let root_dir = get_root_dir();
+        let git = Git::new(&root_dir).unwrap();
+        Releaser { packages, git }
+    }
+
+    pub fn bump(&self, version: Option<Version>) -> Result<()> {
+        ensure_repo_ready(&self.git)?;
+
+        let mut versions: HashMap<String, Version> = HashMap::new();
+        let mut deps: HashMap<PackageType, Vec<String>> = HashMap::new();
+
+        for package in self.packages.iter() {
+            let package_info = package.get_package_info()?;
+            let repo_version = package_info.version;
+
+            versions.insert(package.name.clone(), repo_version);
+            deps.entry(package.package_type)
+                .or_default()
+                .push(package.name.clone());
+        }
+
+        let bump_version = if let Some(version) = version {
+            version
+        } else if let Some(max_version) = versions.values().max() {
+            prompt_bump_version(max_version.clone())
+        } else {
+            anyhow::bail!("No packages found to determine the version.");
+        };
+
+        for package in self.packages.iter() {
+            let old_version = &versions[&package.name];
+            if old_version < &bump_version {
+                println!(
+                    "Bumping {} package {} from {} to {}",
+                    style(&package.package_type).bold().red(),
+                    style(&package.name).bold().cyan(),
+                    style(&old_version).bold().magenta(),
+                    style(&bump_version).bold().green()
+                );
+                package.set_repo_version(&bump_version)?;
+            } else {
+                println!(
+                    "{} package {} is already at {}",
+                    style(&package.package_type).bold().red(),
+                    style(&package.name).bold().cyan(),
+                    style(&bump_version).bold().yellow()
+                );
+            }
+        }
+
+        // Bump dependencies
+        for package in self.packages.iter() {
+            let package_type = package.package_type;
+            let Some(deps) = deps.get(&package_type) else {
+                continue;
+            };
+            println!(
+                "Updating dependencies for {} package {}",
+                style(&package.package_type).bold().red(),
+                style(&package.name).bold().cyan()
+            );
+            let result = package
+                .package_type
+                .get_package_manager()
+                .update_dependencies(&package.path, deps, &bump_version)
```

Comment:
We also need to update `debugger/package.json`, `examples/nextjs/package.json`, and `examples/vanilla/package.json`